pkg/types/foliage.go

```diff
@@ -6,47 +6,44 @@ import (
-// @TODO Streamable
 type FoliageBlockData struct {
-	UnfinishedRewardBlockHash Bytes32              `json:"unfinished_reward_block_hash"`
-	PoolTarget                PoolTarget           `json:"pool_target"`
-	PoolSignature             mo.Option[G2Element] `json:"pool_signature"`
-	FarmerRewardPuzzleHash    Bytes32              `json:"farmer_reward_puzzle_hash"`
-	ExtensionData             Bytes32              `json:"extension_data"`
+	UnfinishedRewardBlockHash Bytes32              `json:"unfinished_reward_block_hash" streamable:""`
+	PoolTarget                PoolTarget           `json:"pool_target" streamable:""`
+	PoolSignature             mo.Option[G2Element] `json:"pool_signature" streamable:""`
+	FarmerRewardPuzzleHash    Bytes32              `json:"farmer_reward_puzzle_hash" streamable:""`
+	ExtensionData             Bytes32              `json:"extension_data" streamable:""`
 }
 
 // Foliage Foliage
 // https://github.com/Chia-Network/chia-blockchain/blob/main/chia/types/blockchain_format/foliage.py#L52
-// @TODO Streamable
 type Foliage struct {
-	PrevBlockHash                    Bytes32              `json:"prev_block_hash"`
-	RewardBlockHash                  Bytes32              `json:"reward_block_hash"`
-	FoliageBlockData                 FoliageBlockData     `json:"foliage_block_data"`
-	FoliageBlockDataSignature        G2Element            `json:"foliage_block_data_signature"`
-	FoliageTransactionBlockHash      mo.Option[Bytes32]   `json:"foliage_transaction_block_hash"`
-	FoliageTransactionBlockSignature mo.Option[G2Element] `json:"foliage_transaction_block_signature"`
+	PrevBlockHash                    Bytes32              `json:"prev_block_hash" streamable:""`
+	RewardBlockHash                  Bytes32              `json:"reward_block_hash" streamable:""`
+	FoliageBlockData                 FoliageBlockData     `json:"foliage_block_data" streamable:""`
+	FoliageBlockDataSignature        G2Element            `json:"foliage_block_data_signature" streamable:""`
+	FoliageTransactionBlockHash      mo.Option[Bytes32]   `json:"foliage_transaction_block_hash" streamable:""`
+	FoliageTransactionBlockSignature mo.Option[G2Element] `json:"foliage_transaction_block_signature" streamable:""`
 }
 
 // FoliageTransactionBlock foliage transaction block
 // https://github.com/Chia-Network/chia-blockchain/blob/main/chia/types/blockchain_format/foliage.py#L29
-// @TODO Streamable
 type FoliageTransactionBlock struct {
-	PrevTransactionBlockHash Bytes32   `json:"prev_transaction_block_hash"`
-	Timestamp                Timestamp `json:"timestamp"`
-	FilterHash               Bytes32   `json:"filter_hash"`
-	AdditionsRoot            Bytes32   `json:"additions_root"`
-	RemovalsRoot             Bytes32   `json:"removals_root"`
-	TransactionsInfoHash     Bytes32   `json:"transactions_info_hash"`
+	PrevTransactionBlockHash Bytes32 `json:"prev_transaction_block_hash" streamable:""`
+	Timestamp                uint64  `json:"timestamp" streamable:""`
```

Comment:
Changing from Timestamp to uint64 might break some things on the RPC side of things, where this type is used, though I'm not sure there's a good way to handle getting Timestamp into the streamable format with reflect, so I suppose this has to happen if we want to support streamable properly here?

Reply:
I attempted to use a streamable tag input to create a new type, but it didn't work. Therefore, I believe we need to make that adjustment for it to work with  streamable.

Reply:
Sounds like we're going to need to use the streamable tag to note types for the CLVM types as well, so if we can make that work, lets flip back to Timestamp here too